### PR TITLE
Code spike for managed stripe accounts

### PR DESCRIPTION
**Pull request by begedin**
# What's in this PR?

This is a code spike for the switch to managed stripe accounts

What we have for now:

## The process

1. Instead of a stripe connect button, we get a form to specify business information. 
2. Submitting a form attempts to save a stripe-connect-account record to the server
3. Server, if stripity stripe uses the latest commit from https://github.com/code-corps/stripity_stripe/pull/153, will create a connect account.

## Related infrastructure, but not hooked into the process yet.

* The "bank account information" form is now also rendered and the action, with the parameters,
is passed all the way to the controller, where it does nothing as of yet.
* All the tests have been fixed.
* A bunch of todos was added indicating how we should continue
* Email is bound from the current user
* I tried to make this as flat and self-contained as possible, so we can merge and split out individual tasks as much as we can.

## References
* Paired with https://github.com/code-corps/code-corps-api/pull/552

### CODE REVIEW

**Comment by JoshSmith** [comment]
I might recommend not doing much at the controller level here and instead just write out the basic form fields per #853. Then we can slowly take that work and use it to break out into separate components in separate PRs, then actual integrated logic after that.

What do you think @begedin?

**Comment by begedin** [comment]
@JoshSmith I got this as far as I could in the time I had. I tried to make it as easy to take over as possible. I believe it can be merged mostly as is and then the work that follows can be performed in smaller chunks.

**Comment by rileytaylor** [comment]
@begedin @JoshSmith Merging and creating some issues for smaller chunks would be fantastic, I'd definitely be able to take a small chunk with the time I have right now.

Routing looks good, basic form setup checks out with the plan. I'm a fan of that pikaday component for date-picking.

**Comment by JoshSmith** [comment]
Yeah as soon as I'm back from the dentist I will break it down.

File: app/controllers/project/settings/donations/payments.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  actions: {
+    onRecipientInformationSubmitted(organization, email, recipientInformation) {
+      let accountParams = merge(recipientInformation, { organization, email });
+      get(this, 'store').createRecord('stripe-connect-account', accountParams)
+                        .save();
+    },
+
+    onBankAccountInformationSubmitted(/* bankAccountInformation */) {
+      // TODO: Handle receiving bank account information
```
Comment (JoshSmith) [inline app/controllers/project/settings/donations/payments.js]: Needs issue.

File: app/templates/components/payments/account-setup.hbs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<h3>Create a Stripe account for {{organizationName}}</h3>
+
+{{!
+  TODO:
+  Account setup should decide if it should render the "form"
+  for each section, or just a generic "section-done" component.
+
+  Possibly, even, the payments controller should decide and then
+  pass in flags to the account setup, but that might be overkill.
```
Comment (JoshSmith) [inline app/templates/components/payments/account-setup.hbs]: Needs issue.

File: app/templates/components/payments/funds-recipient.hbs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  <div class="input-group">
+    <label for="city">City</label>
+    {{input type="text" name="city" value=city}}
+  </div>
+  <div class="input-group">
+    <label for="state">State</label>
+    {{! TODO: Make a proper select dropdown}}
+    <select disabled=true value=state name="state">
+      <option value="CA">California</option>
+    </select>
```
Comment (JoshSmith) [inline app/templates/components/payments/funds-recipient.hbs]: Needs issue.

File: app/templates/components/payments/funds-recipient.hbs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  <div class="input-group">
+    <label for="zip">ZIP</label>
+    {{input type="text" name="zip" value=zip}}
+  </div>
+  <div class="input-group">
+    <label for="country">Country</label>
+    {{! TODO: Make a proper select dropdown}}
+    <select disabled=true value=country name="country">
+      <option value="US">United States</option>
+    </select>
```
Comment (JoshSmith) [inline app/templates/components/payments/funds-recipient.hbs]: Needs issue.

File: tests/integration/components/payments/account-setup-test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+test('it renders', function(assert) {
+  page.render(hbs`
+    {{payments/account-setup
+      onRecipientInformationSubmitted=onRecipientInformationSubmittedHandler
+      onBankAccountInformationSubmitted=onBankAccountInformationSubmittedHandler}}
+  `);
+  assert.equal(this.$('.account-setup').length, 1, 'Component renders');
+});
+
+// TODO: Write tests, remove 'it renders' test
```
Comment (JoshSmith) [inline tests/integration/components/payments/account-setup-test.js]: Needs issue.

File: tests/integration/components/payments/bank-account-test.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    page.removeContext();
+  }
+});
+
+test('it renders', function(assert) {
+  page.render(hbs`{{payments/bank-account bankAccountInformationSubmitted=bankAccountInformationSubmittedHandler}}`);
+  assert.equal(this.$('.bank-account').length, 1, 'Component renders');
+});
+
+// TODO: Write tests, remove 'it renders' test
```
Comment (JoshSmith) [inline tests/integration/components/payments/bank-account-test.js]: Needs issue.

File: tests/integration/components/payments/contact-info-test.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  }
+});
+
+test('it renders email', function(assert) {
+  assert.expect(1);
+  page.render(hbs`{{payments/contact-info email='test@example.com'}}`);
+  assert.equal(page.email, 'test@example.com', 'The email is rendered');
+});
+
+// TODO: Write tests, remove 'it renders' test
```
Comment (JoshSmith) [inline tests/integration/components/payments/contact-info-test.js]: Needs issue.

File: tests/integration/components/payments/funds-recipient-test.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    page.removeContext();
+  }
+});
+
+test('it renders', function(assert) {
+  page.render(hbs`{{payments/funds-recipient}}`);
+  assert.equal(this.$('.funds-recipient').length, 1, 'Component renders');
+});
+
+// TODO: Write tests, remove 'it renders' test
```
Comment (JoshSmith) [inline tests/integration/components/payments/funds-recipient-test.js]: Needs issue.

**Pull request merged**